### PR TITLE
Added usertag and robust regex for diceroll

### PR DESCRIPTION
**Pull request by tobiaskluth**
The testcases might be a bit extensive, but i needed to validate the testregex with sufficient rolls to get critresults.

### CODE REVIEW

**Comment by bbaker2** [comment]
No fault of your own, I failed to update `master` with some important work from `dev`. 

I updated `master` and now you need to re-merge. Sorry about that.

File: src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java [hunk 1]
```diff
@@ -175,6 +179,31 @@ private int getInt(String val) throws BadFormatException {
...
+                case "black":
+                    colour = matcher.group(1);
+                    face = matcher.group(2);
+                    break;
+                default:
+                    colour = matcher.group(2);
+                    face = matcher.group(1);
+            }
+
+            System.out.println(colour + "--> " + face);
```
Comment (bbaker2) [inline src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java:201]: So I'm trying to remove `System.out.println` from the code. I'm aware that my code still has two pre-existing `System.out.println` and I will be removing them soon. 

Please do the same

File: src/test/java/redmarket/commands/roll/RedMarketCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    public void setup() {
+        commandHandler = mock(CommandHandler.class);
+        api = mock(DiscordApi.class);
+        cmd = new RedMarketCommand(commandHandler);
+
+        validRollRgx = Pattern.compile(USER.getMentionTag() + ": black: \\d+ red: \\d+ \\`(Crit)?(Fail|Success)\\` Net: -?\\d+");
+    }
+
+    @Test
+    void testOnRoll() {
```
Comment (bbaker2) [inline src/test/java/redmarket/commands/roll/RedMarketCommandTest.java:37]: You can keep this as is, but after it merges I will be converting it into a parametrized test. 

If you have the time and want to learn something new, checkout https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources

This is an excellent use case for ```
@ParameterizedTest
@ValueSource(strings = { "!r 1b ", "!r 2black", "!r 3r", "etc...." })```

File: src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java [hunk 3]
```diff
@@ -17,20 +18,24 @@
...
+    private final Pattern diceMessageRgx;
     private final Pattern diceFaceRgx;
     private final Pattern modRgx;
 
     private final CommandHandler commandHandler;
 
     public RedMarketCommand(CommandHandler commandHandler) {
         this.commandHandler = commandHandler;
-        diceFaceRgx = Pattern.compile("(black|b|red|r)(\\d+)");
+        diceMessageRgx = Pattern.compile("((black|b|red|r)\\d+|\\d+(black|b|red|r))");
```
Comment (bbaker2) [inline src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java:29]: Running regex twice for almost the exact same pattern seems wasteful. But you're also fixing this bug before I am so I'll probably allow it and eventually make a "performance enhancement" to get this back down to one regex.

File: src/test/java/redmarket/commands/roll/RedMarketCommandTest.java (hunk 2 above)
Comment (bbaker2) [reply src/test/java/redmarket/commands/roll/RedMarketCommandTest.java:37]: But I also want to complement the robust set of tests here. Very good.

**Review by bbaker2 — CHANGES_REQUESTED** [review]
You'll need to rebase or re-merge thanks to my `dev` vs. `master` mistake, but the only real change that I'm going to demand is the removal of `System.out.println`

File: src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java (hunk 1 above)
Comment (tobiaskluth) [reply src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java]: No problem :+1:

File: src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java (hunk 3 above)
Comment (tobiaskluth) [reply src/main/java/com/bbaker/discord/redmarket/commands/roll/RedMarketCommand.java:29]: Yes im aware of that. I read the matcher could handle nested capture groups, but unfortunately I couldn't make it work. I had trouble with the two possibilities canceling eachother out

**Comment by bbaker2** [comment]
I'm a bit of a git history purist. Please drop the `eb5721e Merge branch 'issue2-reference-player-in-roll` and `770c17b removed duplicated function` commits. You can run the following commands and submit again. 
```git checkout -b fixup 199da52
git cherry-pick 9bb0b03
git push -f origin HEAD:issue2-reference-player-in-roll```

**Comment by bbaker2** [comment]
Manually merged so that I could preform a `--ff-only`

**Pull request closed without merging**